registry/client/errors.go

```diff
@@ -10,6 +11,10 @@ import (
 	"github.com/docker/distribution/registry/api/errcode"
 )
 
+// ErrNoErrorsInBody is returned when a HTTP response body parses to an empty
+// errcode.Errors slice.
+var ErrNoErrorsInBody = errors.New("no error details found in HTTP response body")
```

Comment:
> Maybe prefix with "registry/client: ..."

This error is only returned wrapped in a `UnexpectedHTTPResponseError` so it would end up looking like:

```
error parsing HTTP 404 response body: registry/client: no error details found in HTTP response body: ...
```

Maybe add the `registry/client: ` prefix to `UnexpectedHTTPResponseError` and `UnexpectedHTTPStatusError` instead?